### PR TITLE
openshift-metrics: adding duration and resolution options

### LINKED ISSUE #1767: Allow configuration of more options in openshift-metrics
The options ```METRIC_DURATION``` and ```METRIC_RESOLUTION``` should be configurable to allow better control of the hawkular-metrics.

#1582 is a reference to this issue.

### PR DESCRIPTION
**Pull request by enoodle**
Fixes #1767

### CODE REVIEW

**Comment by enoodle** [comment]
@simon3z
I am not sure what should be the default values of duration and resolution, I put in 7 as a place holder.

File: roles/openshift_facts/library/openshift_facts.py [hunk 1]
```diff
@@ -1646,6 +1646,8 @@ def get_defaults(self, roles, deployment_type):
                     deploy=False,
                     storage=dict(
                         kind=None,
+                        duration=7,
```
Comment (jdnieto) [inline roles/openshift_facts/library/openshift_facts.py]: duration and resolution options should be at the same level that deploy or storage

File: roles/openshift_metrics/tasks/main.yaml [hunk 2]
```diff
@@ -43,7 +43,8 @@
   shell: >
    {{ openshift.common.client_binary }} process -f \
    /usr/share/openshift/examples/infrastructure-templates/{{ hawkular_type }}/metrics-deployer.yaml -v \
-    HAWKULAR_METRICS_HOSTNAME=hawkular-metrics.{{ openshift.master.default_subdomain }},USE_PERSISTENT_STORAGE={{ hawkular_persistence }} | \
+    HAWKULAR_METRICS_HOSTNAME=hawkular-metrics.{{ openshift.master.default_subdomain }},USE_PERSISTENT_STORAGE={{ hawkular_persistence }} \
+    METRIC_DURATION=openshift_hosted_metrics_duration,METRIC_RESOLUTION=openshift_hosted_metrics_resolution | \
```
Comment (jdnieto) [inline roles/openshift_metrics/tasks/main.yaml]: Should be METRIC_DURATION={{ openshift.hosted.metrics.duration }}
The tamplate doesn't support yet METRIC_RESOLUTION see [metrics-deployer.yaml](https://github.com/jdnieto/openshift-ansible/blob/ea80a6cb7fc4309e5c5c4210b2ce8504e8008c56/roles/openshift_examples/files/examples/v1.2/infrastructure-templates/enterprise/metrics-deployer.yaml)

File: roles/openshift_facts/library/openshift_facts.py (hunk 1 above)
Comment (enoodle) [inline roles/openshift_facts/library/openshift_facts.py]: Yes, Thanks!

File: roles/openshift_metrics/tasks/main.yaml (hunk 2 above)
Comment (enoodle) [inline roles/openshift_metrics/tasks/main.yaml]: Thanks! Ill see what I can do to add support for it.

**Comment by simon3z** [comment]
> @simon3z
> I am not sure what should be the default values of duration and resolution, I put in 7 as a place holder.

@enoodle default for duration is 7 (days) and for resolution we have to see what @mwringe is going to use.
(I think it's 30, but double check).

**Comment by enoodle** [comment]
This is still WIP as I need to test it correctly add add support for METRIC_RESOLUTION.

**Comment by mwringe** [comment]
@simon3z The default metric resolution for Heapster upstream is 30 seconds, for the previous version we were using it was 10 seconds. The latest in origin-metrics is set to use 10 seconds (to make it more backwards compatible with what we had before). We may set this to 30 seconds though as having this value too low can cause performance issues.

**Comment by simon3z** [comment]
> @simon3z The default metric resolution for Heapster upstream is 30 seconds, for the previous version we were using it was 10 seconds. The latest in origin-metrics is set to use 10 seconds (to make it more backwards compatible with what we had before). We may set this to 30 seconds though as having this value too low can cause performance issues.

@mwringe are you sure about 10 seconds? I think it was 5. We had 4 samples every 20 seconds in CFME, which means that the resolution was 5 seconds in the past.

**Comment by mwringe** [comment]
@simon3z Ah, maybe it was 5 seconds in the past. For CFME would you need this to still be specified as 5 seconds? I can easily change the default in our template to go back to 5 seconds.

**Comment by simon3z** [comment]
@mwringe we have 20 seconds samples so anything higher that that is problematic (e.g. 30 seconds).

**Comment by detiber** [comment]
@sosiouxme @mwringe Which versions of the metrics deployer support METRICS_RESOLUTION and METRICS_DURATION?

**Comment by mwringe** [comment]
@detiber : 
METRICS_DURATION is supported by all versions of origin metrics and OSE Metrics

METRICS_RESOLUTION is much newer (as in less than a week old) and is only supported in the more recent origin-metrics v1.2.0-rc1

File: roles/openshift_metrics/tasks/main.yaml (hunk 2 above)
Comment (enoodle) [inline roles/openshift_metrics/tasks/main.yaml]: Apparently it was already done #1776

**Comment by enoodle** [comment]
Changed the default value of METRICS_RESOLUTION to 10. IIUC it is passed as an environment variable so if an earlier version of openshift-ansible is used it wont raise a problem.

**Comment by detiber** [comment]
ok to test

**Comment by detiber** [comment]
aos-ci-test

**Comment by detiber** [comment]
aos-ci-test

**Pull request merged**